vyper/compiler/output.py

```diff
@@ -37,6 +39,31 @@ def build_userdoc(compiler_data: CompilerData) -> dict:
     return compiler_data.natspec.userdoc
 
 
+def build_solc_json(compiler_data: CompilerData) -> str:
+    # request bytecode to ensure the input compiles, through all the
+    # compilation passes.
+    _ = compiler_data.bytecode
```

Comment:
how important is it to enforce this?

eg. I'm debugging a contract/wip and want to quickly send an archive to a colleague

Reply:
well, that was my original idea, but then i thought it might be a UX issue if you are trying to verify a contract but you upload code that doesn't compile

Reply:
an optional kwarg then?

Reply:
verify=True imo, you gotta know what you are doing if u want unverified packages

Reply:
i think it would be a shame to pass on this usecase, it seems useful.
i agree that the default should be strict, ie verify that it compiles.

Reply:
i think it's a bit tricky behavior. i'm not sure what the behavior of `relax=True` should be. should it skip the integrity sum check (which doesn't exist yet for similar reasons)? should it go through semantic analysis and irgen but stop before bytecode? optimization? etc.